### PR TITLE
enable the tf_prefix parameter

### PR DESCRIPTION
**Pull request by piyushk**
`tf_prefix` is being [deprecated](http://www.ros.org/wiki/hydro/Migration#tf2.2BAC8-Migration.Removal_of_support_for_tf_prefix) in ROS Hydro. I would still like to use the `tf_prefix` parameter http://www.ros.org/wiki/robot_state_publisher#Parameters to distinguish between different robots in multi-robot systems.

It seems that the robot_state_publisher code is not using either `tf::resolve` or reading in the `tf_prefix`, so I am a bit confused as to how this was was working pre-Hydro. I remember running some tests a few months ago that worked.

This patch reads the parameter explicitly and sets it using `tf::resolve`. Using this patch, I can run multiple robots in Gazebo without tf tree collisions.

### CODE REVIEW

**Comment by ugocupcic** [comment]
Is there any plan to integrate that in the default robot_state_publisher in hydro?

We're using the tf_prefix from robot_state_publisher a lot: for example publishing the targets to /targets/joint1 while having the robot positions on /joint1 etc... this way we can visualise easily in rviz.

**Comment by advaitjain** [comment]
Would very much appreciate if this pull request can be merged.

**Comment by isucan** [comment]
This looks good to me. @tfoote what do you think?

File: src/joint_state_listener.cpp [hunk 1]
```diff
@@ -56,8 +56,8 @@
   // set publish frequency
   double publish_freq;
   n_tilde.param("publish_frequency", publish_freq, 50.0);
+  n_tilde.param("tf_prefix", tf_prefix_, std::string(""));
```
Comment (tfoote) [inline src/joint_state_listener.cpp]: To emulate previous behavior this should be searchParam

Comment (piyushk) [inline src/joint_state_listener.cpp]: @tfoote: nice catch. I am stuck in the middle of a few things right now. Will update PR later today and test it.

Comment (piyushk) [inline src/joint_state_listener.cpp]: fixed by @fsuarez6

**Comment by piyushk** [comment]
bump

**Comment by isucan** [comment]
bump

**Comment by wjwwood** [comment]
@tfoote can you comment on this one?

**Comment by tfoote** [comment]
This looks good to me.

**Comment by isucan** [comment]
Should this be merged in hydro, or for indigo?

**Comment by tfoote** [comment]
hydro

**Pull request merged**